nunjucks/src/filters.js

```diff
@@ -266,20 +266,40 @@ function random(arr) {
 
 exports.random = random;
 
+/**
+ * Construct select or reject filter
+ *
+ * @param {function(boolean): boolean} testResultFn
+ * @returns {function(array, string, *): array}
+ */
+function getSelectOrReject(testResultFn) {
+  function filter(arr, testName = 'truthy', secondArg) {
+    const context = this;
+    const test = context.env.getTest(testName);
+
+    return arr.filter(function applyToTest(item) {
```

Comment:
This can be any iterable, not necessarily just an array, right? For example, this template:

```jinja2
{{ vals | reject("sameas", "0") | join("") }}
```

with `vals = '012301230123'`

Reply:
Looks like you right. It not clear from Jinja2 docs, but they have code example that used `for..in`, that seems to work for strings aswel.

I will change to `for` loop.

Reply:
But in Jinja2 test they are testing only [against lists](https://github.com/pallets/jinja/blob/6478c22f29bb85b2bae635602148d95df5e5c7ce/tests/test_filters.py#L664-L672).

Reply:
Can we keep only simple case with arrays?

Adding support for strings `select`/`reject` could be next step, if community will request it.

Reply:
I think you could do this rather simply by replacing `arr` with `Array.from(arr)`

Reply:
Something like this would do the trick:

```javascript
function getSelectOrReject(isTrue) {
  return function filter(arr, testName = 'truthy', ...args) {
    const test = this.env.getTest(testName);
    return Array.from(arr).filter(item =>
      (isTrue === test.apply(this, [item].concat(args))));      
  };
}

exports.select = getSelectOrReject(true);
exports.reject = getSelectOrReject(false);
```

Reply:
`Array.from` is failed in Phantom. Is there any way to inject polyfills to tests bundle?

Reply:
Oh, right. Try importing and using `toArray` from `lib.js`

Reply:
Done